### PR TITLE
front: remove set current simulation inputs state stdcm

### LINKED ISSUE #9267: front: remove the state setCurrentSimulationInputs in stdcm
## Description
Refacto the new stdcm page to remove the state currenSimulationInputs

## AC
- [ ] the code is cleaner

### PR DESCRIPTION
**Pull request by clarani**
closes #9267

### CODE REVIEW

**Comment by SharglutDev** [comment]
This PR will probably conflict with this one : https://github.com/OpenRailAssociation/osrd/pull/9106

File: front/src/applications/stdcmV2/components/StdcmConfig.tsx [hunk 1]
```diff
@@ -42,7 +41,6 @@ const StdcmConfig = ({
   launchStdcmRequest,
   retainedSimulationIndex,
   selectedSimulation,
```
Comment (SarahBellaha) [inline front/src/applications/stdcmV2/components/StdcmConfig.tsx]:
```suggestion
```

File: front/src/applications/stdcmV2/components/StdcmConfig.tsx [hunk 2]
```diff
@@ -32,7 +32,6 @@ type StdcmConfigProps = {
   launchStdcmRequest: () => Promise<void>;
   retainedSimulationIndex: number;
   selectedSimulation?: StdcmSimulation;
```
Comment (SarahBellaha) [inline front/src/applications/stdcmV2/components/StdcmConfig.tsx]:
```suggestion
```
It looks like we don't use it anymore

File: front/src/applications/stdcmV2/components/StdcmConfig.tsx [hunk 3]
```diff
@@ -73,7 +71,6 @@ const StdcmConfig = ({
   const inputsProps = {
     disabled,
     selectedSimulation,
```
Comment (SarahBellaha) [inline front/src/applications/stdcmV2/components/StdcmConfig.tsx]:
```suggestion
```
We can then just pass `disable` to `StdcmConsist`, `StdcmOrigin`, `StdcmVias` & `StdcmDestination` :)

**Review by SarahBellaha — APPROVED** [review]
LGTM, i only left tiny comments :)

File: front/src/applications/stdcmV2/components/StdcmConfig.tsx (hunk 2 above)
Comment (clarani) [reply front/src/applications/stdcmV2/components/StdcmConfig.tsx]: Nice catch, done :white_check_mark:

File: front/src/applications/stdcmV2/components/StdcmConfig.tsx (hunk 1 above)
Comment (clarani) [reply front/src/applications/stdcmV2/components/StdcmConfig.tsx]: Done

File: front/src/applications/stdcmV2/components/StdcmConfig.tsx (hunk 3 above)
Comment (clarani) [reply front/src/applications/stdcmV2/components/StdcmConfig.tsx]: Done

**Review by Akctarus — APPROVED** [review]
tested, LGTM

File: front/src/applications/stdcmV2/components/StdcmConsist.tsx [hunk 4]
```diff
@@ -93,26 +93,8 @@ const StdcmConsist = ({ setCurrentSimulationInputs, disabled = false }: StdcmCon
     } else {
       searchRollingStock('');
     }
-    setCurrentSimulationInputs((prevState) => ({
-      ...prevState,
-      consist: {
-        tractionEngine: rollingStock,
-      },
```
Comment (emersion) [inline front/src/applications/stdcmV2/components/StdcmConsist.tsx:100]: Previously, changing the rolling stock would clear any previously set `speedLimitByTag`. Do we want to retain this behavior?

If so, we need to call `dispatchUpdateSpeedLimitByTag()` in the spots where we're calling `updateRollingStockID()`.

**Review by emersion — COMMENTED** [review]
Apart from this question, looks good! Thanks a bunch for cleaning this up :)

File: front/src/applications/stdcmV2/components/StdcmConsist.tsx (hunk 4 above)
Comment (clarani) [reply front/src/applications/stdcmV2/components/StdcmConsist.tsx:100]: You're right, I kept the previous behavior :+1:

**Review by emersion — APPROVED** [review]

**Pull request merged**